api/dataclasses/project/dataclass.py

```diff
@@ -1,14 +1,69 @@
 from dataclasses import dataclass, field
-from typing import List
+from typing import List, TYPE_CHECKING, Optional
 
-from api.dataclasses.enums import RequirementOperator
+from api.dataclasses.enums import RequirementOperator, RequirementsCriteria
+
+if TYPE_CHECKING:
+    from api.dataclasses.student import Student
 
 
 @dataclass
 class ProjectRequirement:
     attribute: int
     operator: RequirementOperator
     value: int
+    criteria: RequirementsCriteria = RequirementsCriteria.SOMEONE
+    num_members_meeting: Optional[int] = None
```

Comment:
Can we call this `num_members_required` or something instead? I misunderstood what this was initially when reading through the `satisfaction_by_students` function.